### PR TITLE
feat(environment): template publication

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Add the possibility to define a publication template on an environment.
If defined we should check the publication template before publication.

Created a EnvironmentPublisher service with his own factory.
The factory will remember previous publishers, for using it during a bulk publication

https://ems-project.github.io/#/./elasticms/environment/environment

### CODE REVIEW

**Review by coppee — DISMISSED** [review]

File: src/Form/Form/EditEnvironmentType.php [hunk 1]
```diff
@@ -22,51 +23,62 @@ class EditEnvironmentType extends AbstractType
...
-                        'class' => 'btn btn-primary btn-sm ',
-                ],
-                'icon' => 'fa fa-save',
-        ]);
     }
 
     public function configureOptions(OptionsResolver $resolver): void
     {
-        /* set the default option value for this kind of compound field */
-        parent::configureOptions($resolver);
```
Comment (theus77) [inline src/Form/Form/EditEnvironmentType.php:69]: Are you sure that it's not required to call the parent's configureOptions fucntion?

File: src/Service/Revision/RevisionService.php [hunk 2]
```diff
@@ -186,6 +186,11 @@ public function getDocumentInfo(EMSLink $documentLink): DocumentInfo
         return new DocumentInfo($documentLink, $this->revisionRepository->findAllPublishedRevision($documentLink));
     }
 
+    public function getDocumentInfoByRevision(Revision $revision): DocumentInfo
```
Comment (theus77) [inline src/Service/Revision/RevisionService.php]: Does the DocumentInfo entity is abel to know if a revision can be published in a specific environment (for overview tables)?

Comment (Davidmattei) [reply src/Service/Revision/RevisionService.php]: Good remark this change has been removed, did not need the documentInfo but the document.

If you want the documentInfo in the publicaton template -> revison.emsLink|emsco_document_info

File: src/Form/Form/EditEnvironmentType.php (hunk 1 above)
Comment (Davidmattei) [reply src/Form/Form/EditEnvironmentType.php:69]: yes, the parent class is Symfony\Component\Form\AbstractType; which has no implementation on configureOptions.

**Review by coppee — APPROVED** [review]

**Pull request merged**